app/_locales/en/messages.json

```diff
@@ -326,7 +326,7 @@
     "message": "Back to All"
   },
   "backgroundPortClosedError": {
-    "message": "Your connection to Metamask was interrupted. Quit your browser and try again. If issue persists, please log a bug report by clicking on Metamask link below"
+    "message": "Your connection to MetaMask was interrupted. Quit your browser and try again. If this issue persists, please log a bug report by clicking on the link below"
```

Comment:
Agreed

Reply:
I'm having a real hard time adding this button
```
["Restart MetaMask" button]``
```

Reply:
@mcmire @jpuri really really having a difficult time adding that Button. Main issue is we can't use onClick in extension, we have to add an event listener and that doesn't seem to the working for this either.

I noticed though that closing the extension window and re-launching it basically restarts the extension after this error. Can we reframe the message to reflect this e.g

`....so try restarting the extension by closing the extension window and re-opening it.`

Reply:
> Main issue is we can't use onClick in extension, we have to add an event listener and that doesn't seem to the working for this either.

What's not working? Like you can't make the button do anything when clicked?

> Can we reframe the message to reflect this e.g
>
> `....so try restarting the extension by closing the extension window and re-opening it.`

I'm not sure that users will understand what that means. Also, closing the extension window may reload the UI process, but it doesn't reload the background process. Wouldn't we need to do that, if the background process fails to start correctly? It's been a while since I've thought about this so it's possible I'm not understanding the problem correctly anymore.

Reply:
> What's not working? Like you can't make the button do anything when clicked?

Yeah, because inline js is not supported. Usually you setup an event listener for a button using the ID or classname and when the button gets clicked, the event listener is fired. In this case, the button is dynamic and the inline-js is not executed.

>  Also, closing the extension window may reload the UI process, but it doesn't reload the background process
In the cases I've simulated, when background process is not properly started you close and re-open the UI and it restarts the background process. Even in MV3, when you open a UI and the service worker is dead, it restarts the service worker.

Reply:
> In the cases I've simulated, when background process is not properly started you close and re-open the UI and it restarts the background process. Even in MV3, when you open a UI and the service worker is dead, it restarts the service worker.

Ah okay. I learned something new today then :)

> In this case, the button is dynamic

You should still be able to attach an event listener to a button that's added dynamically, using `addEventListener`. I think you have to find the button after it's added to the DOM though, say, after here: https://github.com/MetaMask/metamask-extension/pull/14461/files#diff-5a9d964db9d7c2fe207e21470884b2710e54723e0c1e6e6f7fd264706604fd6cR32.

Reply:
Hi @segun sorry for the bump — were you able to look into this?